public/app/plugins/datasource/prometheus/components/VariableQueryEditor.tsx

```diff
@@ -164,15 +171,21 @@ export const PromVariableQueryEditor = ({ onChange, query, datasource }: Props)
     }
   };
 
+  const onLabelNamesMatchChange = (regex: string) => {
+    if (qryType === QueryType.LabelNames) {
+      onChangeWithVariableString({ qryType, match: regex });
+    }
+  };
+
   /**
    * Call onchange for metric change if metrics names (regex) query type
    * Debounce this because to not call the API for every keystroke.
    */
-  const onMetricChange = debounce((value: string) => {
+  const onMetricChange = (value: string) => {
```

Comment:
Was the debounce not working correctly here?

Reply:
It wasn't debouncing, it would just delay each execution 300ms, so three keystrokes would cause 3 queries

Reply:
Good catch!